### PR TITLE
Fix rosetta's geth args to no longer require genesis file

### LINKED ISSUE #175: Bug: geth args issue starting up from published docker img (only for testnet)
Occurs in `v0.8.4-beta` and later, caused by the `--alfajores` flag not being passed into the geth node on starting the Docker container. Then, the mismatching genesis file (testnet genesis file vs. expected mainnet file) causes geth to abort and not start properly.

error from geth logs
```
INFO [06-29|16:14:10.035] Persisted trie from memory database      nodes=51 size=9.91KiB time=120.725µs gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=-116.00B
Fatal: Error starting protocol stack: database contains incompatible genesis (have 4a3821bde05050292544628030281fbb76e6050ef86bc345e751556cc0f85ce2, new e423b034e7f0282c1b621f7bbc1cea4316a2a80b1600490769eae77777e4b67e)
```
Repro via:
```sh
$ export RELEASE="latest"  # or specify a release version
# folder for rosetta to use as data directory (saves rosetta.db & celo-blockchain datadir)
export DATADIR="${PWD}/datadir"
mkdir $DATADIR
curl 'https://storage.googleapis.com/genesis_blocks/alfajores' > "${DATADIR}/genesis.json"
docker pull us.gcr.io/celo-testnet/rosetta:$RELEASE
docker run --name rosetta --rm \
  -v "${DATADIR}:/data" -v "${DATADIR}/logs:/logs"\
  -p 8080:8080 \
  us.gcr.io/celo-testnet/rosetta:$RELEASE \
  run  --geth.syncmode full \
  --geth.gcmode archive
```

Low priority, don't know of many folks using this image in particular, but good idea to fix this as it will cause the same issue for all future builds.

### PR DESCRIPTION
**Pull request by eelanagaraj**
Up til now we have been pulling the genesis file from cloud storage, but this information is now stored in the celo-blockchain repo and we should use that as the source of truth.

closes #175

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ Flags:
       --datadir string            datadir to use
       --geth.binary string        Path to the celo-blockchain binary
       --geth.bootnodes string     Bootnodes to use (separated by ,)
-      --geth.genesis string       path to the genesis.json
+      --geth.genesis string       path to the genesis.json (optional; useful for running Rosetta witha  custom chain)
```
Comment (oneeman) [inline README.md:53]:
```suggestion
      --geth.genesis string       path to the genesis.json (optional; useful for running Rosetta with a custom chain)
```

File: cmd/run.go [hunk 2]
```diff
@@ -76,8 +76,9 @@ func init() {
 	flagSet.String("geth.ipcpath", "", "Path to the geth ipc file")
 	utils.ExitOnError(serveCmd.MarkFlagFilename("geth.ipcpath"))
 
-	flagSet.String("geth.genesis", "", "path to the genesis.json")
+	flagSet.String("geth.genesis", "", "(Optional) path to the genesis.json, for use with custom chains")
 	utils.ExitOnError(serveCmd.MarkFlagFilename("geth.genesis", "json"))
+	flagSet.String("geth.network", "", "Network to use, either 'mainnet', 'alfajores', or 'baklava'")
```
Comment (oneeman) [inline cmd/run.go:81]: This should also be marked a optional, since you specify either this or geth.genesis.

**Review by oneeman — CHANGES_REQUESTED** [review]
Looks good, but I think you could go further in removing things.  When you run celo-blockchain for the first time it doesn't need an empty data directory to exist, it creates it by itself.  So for alfajores, baklava, and mainnet we could remove the `make foo-env` entirely, and remove the bullet point in their docs saying about having to run them.

Besides that, I just commented with a couple of nits.

**Comment by eelanagaraj** [comment]
thanks @oneeman! Looks like Rosetta does some additional datadir creation stuff before even calling geth (i.e. creating the `rosetta.db` at this location). I took out the pieces that were easy to remove and that wouldn't potentially disrupt any other parts of the service, though it still creates the dir before starting up the geth node -- maybe the full changes will be worth it in the future.

**Review by oneeman — APPROVED** [review]
Looks great!

**Pull request merged**